lua/nvim-treesitter/ts_utils.lua

```diff
@@ -337,18 +337,26 @@ function M.memoize_by_buf_tick(fn, options)
   end
 end
 
+local function length_last_line(range)
+  if range.start.line == range["end"].line then
+    return range["end"].character - range.start.character + 1
+  else
+    return range["end"].character + 1
+  end
+end
```

Comment:
I'm not sure whether this is correct. When end col = 0 then get_node_text would not include this last line also the + 1 shouldn't be necessary. As `get_vim_range` shows it is apparently not possible to get the length of the last line without accessing the actual text. Probably you pattern matching approach would be better (or return to ye old get_node_text)

Reply:
My thought with +1 is that ranges are zero-based and end-inclusve? So, for example, if range.end.character is 12, it means it includes characters 0, 1, 2, ..., 12, i.e. 13 characters.

I see what you mean though, range.end.character may be "out of bounds"? (accessing the 0th element in an empty array is an out of bound errors, but it seems that that's expected to work here?)

Are you saying that for a range like `{start = {line = 1, col = 0}, end = {line = 10, col = 0}}`, get_node_text will only return 9 lines? In that case this won't work either: https://github.com/nvim-treesitter/nvim-treesitter/pull/2838/files#diff-ee62282252e6dfba5eebd5b1a5a1ff9df64f01392723b3044965d5c7469e9915R371

Reply:
but yeah it sounds like the best solution for now is to keep nvim-treesitter's version of get_node_text that returns a table with the lines instead of using vim.treesitter.query.get_node_text?

Reply:
@clason are you fine with reverting? Most plugin can do the switch easily while for some application the join/split is just a needless round trip. For swap it's not critical since swap is slow due to the queries finding the swap objects